File: Codeforces/Div3/748div3(vp)/F(补).cpp
#include <bits/stdc++.h>
#define ll long long
#define int ll
using namespace std;
const int N = 55;

int t, n, A, B;
int vis[N][N][N][N];

void init () {memset (vis, 0, sizeof vis);}

void charming () {
	init ();
	cin >> n >> A >> B;
	int mn = INT_MAX;
	string s, res, color;
	cin >> s;
	auto dfs = [&] (auto &&me, int id, int red, int black, int cnt, string &color) {
		if (id >= n) {
			int diff = abs (n - 2 * cnt);
			if (red || black || diff >= mn || cnt < 1 || cnt > n - 1) return;
			mn = diff;
			res = color;
			return;
		}
		if (vis[id][red][black][cnt]) return;
		vis[id][red][black][cnt] = true;
		color.append ("R");
		me (me, id + 1, (red * 10 + s[id] - '0') % A, black, cnt + 1, color);
		color.pop_back ();
		color.append ("B");
		me (me, id + 1, red, (black * 10 + s[id] - '0') % B, cnt, color);
		color.pop_back ();
	};
	dfs (dfs, 0, 0, 0, 0, color);
	if (mn >= INT_MAX) cout << -1 << endl;
	else cout << res << endl;
}

signed main () {
	cin >> t;
	while (t--) charming ();
	return 0;
}

File: Codeforces/Div3/744div3(vp)/F(补).py
import sys
N = int (1e6 + 5)
sys.setrecursionlimit (N)

def charming () :
  n, d = map (int, input ().split ())
  a = list (map (int, input ().split ()))
  vis = [0] * n
  res = 0
  for i in range (n) :
    if vis[i] : continue
    j = i;
    b = list ()
    ok = False
    while vis[j] == 0 :
      if a[j] == 0 : ok = True
      b.append (a[j])
      vis[j] = True
      j = (j + d) % n
    if ok == False :
      print (-1)
      return
    p = len (b) - 1
    while p >= 0 and b[p] == 1 : p -= 1
    cnt = len (b) - p - 1
    for i in b :
      if i == 1 :
        cnt += 1
        res = max (res, cnt)
      else : cnt = 0
  print (res)
      

for t in range (int (input ())) : charming ()
